### ISSUE TITLE
create a ServiceAccount after the CRD applied  the ServiceAccount is not updating by the operator

### ISSUE THREAD

**Issue by developer-guy** [alexellis/registry-creds#12]
**Describe the bug**
While working on PR #10, I noticed that, if we create a ServiceAccount before the CRD applied operator works fine and ServiceAccount updated as it's supposed to be, but If we are creating a ServiceAccount after the CRD applied this reconciler will not update the ServiceAccount because we are not observing ServiceAccount Events.

**To Reproduce**
```sh
Step 1 : $ eval $(minikube docker-env)

Step 2: $ make docker-build

Step 3: $ make install deploy

Step 4: $ arkade install ingress-nginx

Step 5: $ arkade install cert-manager

Step 6: $ kubectl create secret docker-registry registry-creds-secret \
  --namespace kube-system \
  --docker-username=$USERNAME \
  --docker-password=$PW \
  --docker-email=$EMAIL

Step 7: $  echo -n "apiVersion: ops.alexellis.io/v1
kind: ClusterPullSecret
metadata:
  name: dockerhub
spec:
  secretRef:
    name: registry-creds-secret
    namespace: kube-system" | k apply -f -

Step 8: $ kubectl get serviceaccounts default -oyaml  # you notice that "imagePullSecrets" added.

Step 9: $ $ kubectl create serviceaccount demo-sa 

Step 10: $ kubectl get serviceaccounts demo-sa -oyaml # you notice that "imagePullSecrets" not added, but if we'd done that before applying CRD (Step 7.), this would work fine.
```

**Expected behavior**
If we create a ServiceAccount after the CRD applied to the cluster, `imagePullSecrets` should be added to that ServiceAccount.

**Screenshots or console output**

**Operating system and version:**
macOS 10.15.6

**List all possible solutions, and your suggested option**
We need to create an additional reconcile loop for the ServiceAccount that observes the events as we did already at namespace_watcher.go.

**Comment by kingdonb** [comment]
I came here to report I think the same issue, although my report is shaped slightly differently.

I create an NS "nginx-ingress" and install the ingress controller via Helm chart, immediately after setting up my ClusterPullSecret resource. I follow the same step `kubectl get -n default serviceaccounts default -oyaml` and see an imagePullSecrets ref there, `- name: dockerhub-registrycreds` just as expected, but the default service account in my newly created ns `nginx-ingress` is missing the imagePullSecret ref.

Deleting and recreating the ClusterPullSecret resolves it. So basically the same behavior issue as reported above by @developer-guy – new namespaces get the secret copied successfully, but the imagePullSecret reference is not added to the service accounts. My cluster is a Kind 1.19.1 on MacOS, with Docker Desktop 2.5.0.0 backing the cluster.

**Comment by developer-guy** [comment]
hello @kingdonb, we fixed this problem at PR #14, we are waiting for this PR to be approved by Alex, then we will catch the ServiceAccount creation event and update it accordingly.

**Comment by developer-guy** [comment]
@alexellis I think we can close this issue.
